documents/DeveloperGuide/ShaderGeneration.md

```diff
@@ -29,7 +29,7 @@ In the following sub-sections each of these methods are explained. For all metho
 Provided code generators support a very simple expression language for inlining code. This is useful for simple nodes where the operation can be expressed as a single line of code. Inlining will reduce the number of function calls and produce more compact code. The syntax to use is the same as the target shading language, with the addition of using the node’s input ports as variables wrapped in double curly brackets: `{{input}}`. The code generator will replace these variables with values assigned or connected to the respective inputs. Figure 2 gives an example.
 
 Connecting the expression to the nodedef is done using an `<implementation>` element as seen in
-Figure 2. The file extension is used to differentiate inline expressions from source code functions, using `filename.inline`.
+Figure 2. The first option is to keep inline code in a file. The file extension is used to differentiate inline expressions from source code functions, using `filename.inline`. The second option is to directly embed the inlined code using `sourcecode`. This is the recommended approach for inlining if there the logic can fit on one line of code.
```

Comment:
@niklasharrysson , adding in note about `sourcecode` usage. Whatever you think is best to say here let me know, or replace it directly. Thanks.